app/next-client-app/components/scanreports/HandleArchive.tsx

```diff
@@ -0,0 +1,29 @@
+import { archiveDataSets } from "@/api/datasets";
+import { updateScanReport } from "@/api/scanreports";
+import { toast } from "sonner";
+
+export const HandleArchive = async ({
+  id,
+  hidden,
+  ObjName,
+  type,
+}: {
+  id: number;
+  hidden: boolean;
+  ObjName: string;
+  type: string;
+}) => {
+  const message = hidden ? "Unarchive" : "Archive";
+  let response;
+  if (type === "datasets") {
+    response = await archiveDataSets(id, !hidden);
+  } else if (type === "scanreports") {
+    response = await updateScanReport(id, "hidden", !hidden);
+  }
```

Comment:
Because we are not returning any "error" so I don't think try and catch block is needed here. They will show the Toast depending on the string response (if it has the indicator "Error" or not)